build.gradle.kts

```diff
@@ -0,0 +1,155 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import java.net.URI
+import net.ltgt.gradle.errorprone.errorprone
+
+plugins {
+    idea
+    java
+    pmd
+    id("net.ltgt.errorprone") version ("2.0.2")
+}
+
+subprojects {
+    apply {
+        plugin("java")
+        plugin("pmd")
+        plugin("net.ltgt.errorprone")
+
+        from("$rootDir/version.gradle.kts")
+    }
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven {
+            url = URI("https://europe-maven.pkg.dev/spine-event-engine/releases")
+        }
+        maven {
+            url = URI("https://europe-maven.pkg.dev/spine-event-engine/snapshots")
+        }
+    }
+
+    val javaVersion = JavaVersion.VERSION_1_8
+    java {
+        sourceCompatibility = javaVersion
+        targetCompatibility = javaVersion
+    }
+
+    val guavaVersion: String by extra
+    val grpcVersion: String by extra
+
+    val errorProneCoreVersion: String by extra
+    val errorProneJavacVersion: String by extra
+    val checkerFrameworkVersion: String by extra
+
+    val apiGuardianVersion: String by extra
+    val junit5Version: String by extra
+
+    dependencies {
+        implementation("com.google.guava:guava:$guavaVersion")
+        runtimeOnly("io.grpc:grpc-netty:$grpcVersion")
+
+        errorprone("com.google.errorprone:error_prone_core:$errorProneCoreVersion")
+        errorproneJavac("com.google.errorprone:javac:$errorProneJavacVersion")
+        implementation("org.checkerframework:checker-qual:$checkerFrameworkVersion")
+
+        testImplementation("org.junit.jupiter:junit-jupiter-api:$junit5Version")
+        testImplementation("org.junit.jupiter:junit-jupiter-params:$junit5Version")
+        testImplementation("org.apiguardian:apiguardian-api:$apiGuardianVersion")
+        testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:$junit5Version")
+    }
+
+    configurations {
+        all {
+            resolutionStrategy {
+                force(
+                    "com.google.guava:guava:$guavaVersion",
+                    "com.google.truth:truth:1.1.3",
+                    "com.google.truth.extensions:truth-java8-extension:1.1.3",
+                    "com.google.truth.extensions:truth-proto-extension:1.1.3"
+                )
+            }
+        }
+    }
+
+    tasks.withType<JavaCompile> {
```

Comment:
Let's try to make the `build.gradle.kts` much more compact and readable. I would prefer moving the extractable parts of the script (like this one) to `buildSrc`.